src/command_modules/azure-cli-iot/azure/cli/command_modules/iot/_params.py

```diff
@@ -16,16 +16,29 @@ def get_device_id_completion_list(prefix, action, parsed_args, **kwargs):#pylint
     client = iot_hub_service_factory(kwargs)
     return [d.device_id for d in iot_device_list(client, parsed_args.hub_name, top=100)] if parsed_args.hub_name else []
 
-hub_name_type = CliArgumentType(completer=get_resource_name_completion_list('Microsoft.Devices/IotHubs'),
-                                help='IoT Hub name.')
 
-register_cli_argument('iot hub', 'hub_name', hub_name_type, options_list=('--name', '-n'))
-register_cli_argument('iot device', 'hub_name', hub_name_type)
+# Arguments for 'iot' group
+register_cli_argument('iot', 'hub_name', help='IoT Hub name.',
+                      completer=get_resource_name_completion_list('Microsoft.Devices/IotHubs'))
 register_cli_argument('iot', 'device_id', options_list=('--device-id', '-d'), help='Device Id.',
                       completer=get_device_id_completion_list)
 
+# Arguments for 'iot hub consumer-group' group
+register_cli_argument('iot hub consumer-group', 'consumer_group_name', options_list=('--name', '-n'),
+                      help='Event hub consumer group name.')
+register_cli_argument('iot hub consumer-group', 'event_hub_name', help='Target event hub name.')
+
+# Arguments for 'iot hub key' group
+register_cli_argument('iot hub key', 'key_name', options_list=('--name', '-n'), help='Share access policy name.')
+
+# TODO: Find a better way to register options list for below commands
```

Comment:
Nice suggestion. Thanks!

---
In reply to: [87031719](https://github.com/Azure/azure-cli/pull/1263#discussion_r87031719) [](ancestors = 87031719)